src/correctionlib/convert.py

```diff
@@ -59,7 +64,6 @@ def read_axis(axis: "PlottableAxis", pos: int) -> Variable:
     def edges(axis: "PlottableAxis") -> List[float]:
         out = []
         for i, b in enumerate(axis):
-            assert isinstance(b, tuple)
```

Comment:
I'm confused. According to UHI I should still expect even ROOT axes iterators to be `Tuple[float, float]` based on the [spec](https://github.com/scikit-hep/uhi/blob/2d11a1cb621370bf16f8dd3a0bf0e37ac6442793/src/uhi/typing/plottable.py#L95-L99). See [PlottableAxis](https://github.com/scikit-hep/uhi/blob/2d11a1cb621370bf16f8dd3a0bf0e37ac6442793/src/uhi/typing/plottable.py#L95-L99) for all options in the spec.
Ideally we could automatically create Category nodes from `int` or `str` iterates.

Reply:
Or is it that the numpy array of length 2 duck-types as a tuple? Yep.
So to handle the alternative cases now, we should check the inverse:
```python
if isinstance(b, (str, int)):
    raise ValueError("cannot auto-convert string or integer category axes (yet)")
```

Reply:
> Or is it that the numpy array of length 2 duck-types as a tuple? Yep.

Yes, that's what's going on.

Thanks for the suggestion, added. However I'm having a bit of trouble with the mypy checks, which I'm not used to, you probably know how to deal with those...

Reply:
I can take of those, sure